classes/class-p4-campaigns.php

```diff
@@ -222,18 +227,48 @@ public function save_taxonomy_meta( $term_id ) {
 			if ( $this->validate( $attachment_id ) ) {
 				update_term_meta( $term_id, $field_id, $attachment_id );
 				update_term_meta( $term_id, $field_url, $attachment_url );
+				$happy_background_code = ' background="' . $attachment_id . '" ';
+			} else {
+				$happy_background_code = '';
 			}
 
 			$field_id              = 'happypoint_bg_opacity';
 			$happypoint_bg_opacity = filter_input( INPUT_POST, $field_id, FILTER_VALIDATE_INT );
 
 			if ( $this->validate( $happypoint_bg_opacity ) ) {
 				update_term_meta( $term_id, $field_id, $happypoint_bg_opacity );
+			} else {
+				$happypoint_bg_opacity = 30;
 			}
 
 			$redirect_page = $_POST['redirect_page'] ?? 0;
-			if ( ! is_null( get_post( $redirect_page ) ) || ! $redirect_page ) {
+			if ( ! is_null( get_post( $redirect_page ) ) && 0 != $redirect_page ) {
 				update_term_meta( $term_id, 'redirect_page', $redirect_page );
+			} else {
+
+				$tag_data     = get_term( $term_id );
+				$post_content = '[shortcake_newcovers cover_type="1" tags="' . $term_id . '" covers_view="0" title="' . __( 'Things you can do', 'planet4-master-theme' ) . '"  description="' . __( 'We want you to take action because together we\'re strong.', 'planet4-master-theme' ) . '" /]
+
+[shortcake_articles tags="' . $term_id . '" ignore_categories="false" /]
+
+[shortcake_newcovers cover_type="3" title="' . __( 'Publications', 'planet4-blocks' ) . '" tags="' . $term_id . '" post_types="62" covers_view="0" /]
+
+[shortcake_happy_point ' . $happy_background_code . ' opacity="' . $happypoint_bg_opacity . '"/]';
+
+				$my_post = array(
+					'post_title'   => '#' . $tag_data->name,
+					'post_content' => $post_content,
+					'post_status'  => 'publish',
+					'post_type'    => 'page',
+				);
+
+				// Insert the post into the database.
+				$tag_page_id = wp_insert_post( $my_post );
```

Comment:
A check for return type (if it's an int) would be good here.
Insert post could return WP_Error if it fails for some reason.